### ISSUE TITLE
[CURATOR-555] Curator Client leaks the com.google.guava:guava regardless of it being shaded.

### ISSUE THREAD

**Issue by jira-importer** [apache/curator#1074]
<p>The curator-client leaks the <tt>com.google.guava:guava</tt> regardless of it being shaded within the artifact.</p>

<div class="code panel" style="border-width: 1px;"><div class="codeHeader panelHeader" style="border-bottom-width: 1px;"><b><a href="https://search.maven.org/remotecontent?filepath=org/apache/curator/curator-client/4.2.0/curator-client-4.2.0.pom" class="external-link" target="_blank" rel="nofollow noopener">https://search.maven.org/remotecontent?filepath=org/apache/curator/curator-client/4.2.0/curator-client-4.2.0.pom</a></b></div><div class="codeContent panelContent">
<pre class="code-xml">
<span class="code-tag"><dependency></span>
    <span class="code-tag"><groupId></span>com.google.guava<span class="code-tag"></groupId></span>
    <span class="code-tag"><artifactId></span>guava<span class="code-tag"></artifactId></span>
<span class="code-tag"></dependency></span> 
</pre>
</div></div>

<p>Note that since <tt>curator-client</tt> 4.2.0 was moved to Guava 27.0.1 as pat of <a href="https://issues.apache.org/jira/browse/CURATOR-502" title="Update dependency com.google.guava:guava of org.apache.curator:curator-client" class="issue-link" data-issue-key="CURATOR-502"><del>CURATOR-502</del></a> this can cause problems for projects which depend on older versions of Guava but get upgraded by the transitive dependency to 27.0.1.</p>

---
<details><summary><i>Originally reported by <a title="Bernardo Gomez Palacio" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=berngp">berngp</a>, imported from: <a href="https://issues.apache.org/jira/browse/CURATOR-555" target="_blank">Curator Client leaks the com.google.guava:guava regardless of it being shaded.</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>resolution</b>: Unresolved
<li><b>imported</b>: 2025-01-21
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=berngp">berngp</a>:</i>
<p>Based on <a href="https://cwiki.apache.org/confluence/display/CURATOR/TN13" class="external-link" target="_blank" rel="nofollow noopener">https://cwiki.apache.org/confluence/display/CURATOR/TN13</a> it looks like this will always be the case. Users can pin to the specific version they need/want.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=randgalt">randgalt</a>:</i>
<p>Correct per that Tech Note. At some point, we can bite the bullet and make the breaking change.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kezhuw">kezhuw</a>:</i>
<p>Ideally, <a href="https://issues.apache.org/jira/browse/CURATOR-558" title="ZooKeeper 3.6.0 has many API changes - bring Curator up to date" class="issue-link" data-issue-key="CURATOR-558"><del>CURATOR-558</del></a> should unblock this in upgrading to 5.5.0. But <a href="https://issues.apache.org/jira/browse/CURATOR-670" title="CuratorTransactionResult.ofTypeAndPath leaks guava Predicte to public" class="issue-link" data-issue-key="CURATOR-670">CURATOR-670</a> leaks guava to public still. To solve this, we have to either break something or postpone to next major release.</p>

**Comment by jira-importer** [comment]
<i>[Originally related to: <a href="https://github.com/apache/curator/issues?q=in%3Atitle%20CURATOR-702">CURATOR-702</a>]</i>
